backend/__tests__/userModel.test.js

```diff
@@ -52,27 +49,72 @@ describe("Volunteer User", () => {
 });
 
 describe("Signup API", () => {
-  it("saves a user to the database", async () => {
-    const res = await createUser();
-    const createdUser = await User.findOne({email:  'oxfam@oxfam.com'})
-    expect(res.status).toBe(200)
-    expect(createdUser._id).toBeDefined()
-  })
+    it("saves a user to the database", async () => {
+        const res = await createUser();
+        const createdUser = await User.findOne({email:  'oxfam@oxfam.com'})
+        expect(res.status).toBe(200)
+        expect(createdUser._id).toBeDefined()
+    })
 })
 
-
 describe("Login", () => {
-    xit("A user can log in with the correct password", async () => {
+    it("A user can log in with the correct password", async () => {
         let user = await createUser();
-        // console.log("User", user.charityName)
         let userLoggedIn = await logInUser();
-        // console.log("UserLoggedIn", userLoggedIn)
         let parsedUser = JSON.parse(user.text);
-        console.log("Parsed User", parsedUser)
         let parsedUserLoggedIn = JSON.parse(userLoggedIn.text);
-        // console.log("parsedUserLoggedIn", parsedUserLoggedIn)
         expect(userLoggedIn.status).toEqual(200);
-        expect(parsedUser.name).toBe(wesleyData.name);
-        expect(parsedUserLoggedIn.user).toEqual("wesley@example.com");
+        expect(parsedUser.charityName).toBe("Oxfam");
+        expect(parsedUserLoggedIn.user).toEqual("oxfam@oxfam.com");
+    });
+
+
+    it("A user cannot log in with an incorrect password", async () => {
+        await request
```

Comment:
Do we need to actually create a user here first? Isn't the DB cleared between tests, so there will not actually be a user with this info until we make one? Which means that this is actually testing for trying to log-in as a non-existent user?

Reply:
This is testing that a signed up user needs the correct password to sign in though